filament/src/details/MaterialInstance.cpp

```diff
@@ -73,7 +73,7 @@ FMaterialInstance::FMaterialInstance(FEngine& engine,
 
     if (!material->getSamplerInterfaceBlock().isEmpty()) {
         mSamplers = other->getSamplerGroup();
-        mSbHandle = driver.createSamplerGroup(mSamplers.getSize());
+        mSbHandle = driver.createSamplerGroup(mSamplers.getSize(), mMaterial->getName().c_str());
```

Comment:
that would be extremely unsafe because the material could be destroyed before the backend gets the pointer.

Reply:
I can replace this with something generic like "User material", but I think it would be super helpful if our error message was able to print out the name of the material. But making that safe requires a string copy and heap allocation. What do you think?

Reply:
What about we use a fixed size for the name, like 32 bytes (which is the size of a std::string). You can then use a `std::array<char*, 32>` 
You could even create a `FixedSizeString<size_t>` in libutils if you wanted to, to make it much more conveniant.